test/e2e/topology_updater.go

```diff
@@ -41,35 +44,154 @@ import (
 var _ = ginkgo.Describe("[TopologyUpdater][InfraConsuming] Node topology updater", func() {
 	var (
 		initialized         bool
+		nodeName            string
+		namespace           string
+		kubeletConfig       *kubeletconfig.KubeletConfiguration
 		topologyClient      *topologyclientset.Clientset
 		topologyUpdaterNode *v1.Node
-		kubeletConfig       *kubeletconfig.KubeletConfiguration
+		workerNodes         []v1.Node
 	)
 
 	f := framework.NewDefaultFramework("topology-updater")
-	ns := getNamespaceName()
 
 	ginkgo.BeforeEach(func() {
 		var err error
 
 		if !initialized {
+			nodeName = getNodeName()
+			namespace = getNamespaceName()
+
 			topologyClient, err = topologyclientset.NewForConfig(f.ClientConfig())
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-			topologyUpdaterNode, err = f.ClientSet.CoreV1().Nodes().Get(context.TODO(), getNodeName(), metav1.GetOptions{})
+			topologyUpdaterNode, err = f.ClientSet.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			workerNodes, err = utils.GetWorkerNodes(f)
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 			initialized = true
 		}
 
-		// intentionally get every single time
+		// intentionally done once
 		kubeletConfig, err = e2ekubelet.GetCurrentKubeletConfig(topologyUpdaterNode.Name, "", true)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
 
 	ginkgo.Context("with cluster configured", func() {
+		ginkgo.It("it should not account for any cpus if a container doesn't request exclusive cpus", func() {
+			ginkgo.By("getting the initial topology information")
+			initialNodeTopo := getNodeTopology(topologyClient, topologyUpdaterNode.Name, namespace)
+			ginkgo.By("creating a pod consuming the shared pool")
+			sleeperPod := &v1.Pod{
```

Comment:
a parametric `makePod` function didn't turn to be so helpful in the kube tests; because of the parametric nature the readability ended up a bit worse. But I guess we can try again.

Reply:
Oh okay, I saw them being heavily used when I worked on integration tests with SIG scheduling that is why recommended that. Thanks for making the change.